core/src/main/java/org/apache/iceberg/actions/BaseRewriteDataFilesAction.java

```diff
@@ -268,7 +272,7 @@ private void replaceDataFiles(Iterable<DataFile> deletedDataFiles, Iterable<Data
     }
   }
 
-  protected abstract FileIO fileIO();
+  protected abstract FileIO setFileIO();
```

Comment:
In the construction method of BaseRewriteDataFilesAction, we need to initialize fileIO and provide an abstract setter method. Other places, such as the rewriteDataForTasks method of spark and flink, need to get fileIO by call the getter method of fileIO, but there is no getter method currently. If we also call the fileIO() method to get fileIO, then the fileIO() method will be called multiple times, and SparkUtil.serializableFileIO() will be called multiple times.

We provide a setFileIO() method to ensure that the initialization of fileIO is only executed once. If we want to get the value of the fileIO field, call its getter method fileIO().

Reply:
I understand your points. I said it's confusing because its method name is `setFileIO` while would just return the `FileIO` without accepting any arguments,   it looks like a `getFileIO`. 

How about loading the fileIO lazily as the previous abstraction PR comment ? 

```java
private FileIO fileIO;

private FileIO fileIO(){
    if(fileIO==null){
       this.fileIO = SparkUtil.serializableFileIO(table());
    } 
    return this.fileIO;
}

```